tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -669,6 +704,93 @@ TEST_P(DistributedTransformerTest, MLP_Layer) {
   validate(expected_outputs, outputs, {0.01, 0.01, 0.02, 0.02});
 }
 
+TEST_P(DistributedTransformerTest, Sequence_Parallel_MLP_Layer) {
+  // TODO: Reshapes that form device axes when D=1 get optimized away causing
+  // failures. This won't be a problem after
+  // https://github.com/NVIDIA/Fuser/issues/2563.
+  if (D == 1) {
+    GTEST_SKIP() << "Requires >1 devices, D=" << D;
+  }
+  if ((4 * E) % D != 0) {
+    GTEST_SKIP() << "Requires number of devices=" << D
+                 << " evenly divide 4*E=" << 4 * E;
+  }
+  if ((B * S) % D != 0) {
+    GTEST_SKIP() << "Requires number of devices=" << D
+                 << " evenly divide B*S=" << B * S;
+  }
+  DataType dtype = GetParam();
+  at::ScalarType at_dtype = data_type_to_aten(dtype);
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  const auto mesh = DeviceMesh::createForNumDevices(D);
+
+  TensorView* x = makeContigConcreteTensor({D, B * S / D, E}, dtype);
```

Comment:
(Not required for this PR)

Did you try `{D, B, S/D, E}`, which is more readable and closer to how the model is built conceptually?

Reply:
I did try this, but since the sequence dimension isn't the outer-most dimension it breaks the analysis in the pointwise scheduler that assumes DID axes are outermost axes. I didn't bother modifying that scheduler since it will have to be fixed for loop + DID parallelization.